csharp-mode.el

```diff
@@ -2077,1076 +1837,282 @@ to the beginning of the prior namespace.
+                        ;; unless we are super-specific and expect the accesors,
+                        ;; lots of weird things gets slurped into the name.
+                        ;; including the accessors themselves.
+                        (regexp-opt '("get" "set"))
+                        ) 1)
+          ;; adding fields... too much?
+          (list "field"
+                (concat bol
+                        access-modifier-list "+"
+                        ;; fields can be readonly/const
+                        "\\(?:" (regexp-opt '("readonly" "const")) space "\\)?"
+                        "\\("
+                        return-type space
+                        generic-identifier
+                        "\\)"
+                        optional-space
+                        ;; optional assignment
+                        "\\(?:=[^;]+\\)?"
+                        ";") 1)
+          (list "indexer"
+                (concat bol
+                        access-modifiers
+                        return-type space
+                        "this" optional-space
+                        "\\("
+                        ;; opening bracket
+                        "\\[" optional-space
+                        ;; type
+                        "\\([^\]]+\\)" optional-space
+                        identifier
+                        ;; closing brackets
+                        "\\]"
+                        "\\)"
+                        optional-space "{" optional-space
+                        ;; unless we are super-specific and expect the accesors,
+                        ;; lots of weird things gets slurped into the name.
+                        ;; including the accessors themselves.
+                        (regexp-opt '("get" "set"))) 1)
+          (list "event"
+                (concat bol
+                        access-modifier-list "+"
+                        optional-space "event" optional-space
+                        "\\("
+                        return-type space
+                        generic-identifier
+                        "\\)"
+                        optional-space
+                        ";") 1))))
+
+(defun csharp--imenu-get-pos (pair)
+  (let ((pos (cdr pair)))
+    (if (markerp pos)
+        (marker-position pos)
+      pos)))
+
+(defun csharp--imenu-get-container (item containers previous)
+  (if (not containers)
+      previous
+    (let* ((item-pos (csharp--imenu-get-pos item))
+           (container (car containers))
+           (container-pos (csharp--imenu-get-pos container))
+           (rest      (cdr containers)))
+      (if (and container-pos
+               (< item-pos container-pos))
+          previous
+        (csharp--imenu-get-container item rest container)))))
+
+(defun csharp--imenu-get-container-name (item containers)
+  (let ((container (csharp--imenu-get-container item containers nil)))
+    (if (not container)
+        nil
+      (let* (;; namespace
```

Comment:
Another unneeded `let*` :D  The rule for these is simple, if you need to make use of the value of a previous binding, use `let*`, otherwise `let`.  Comments could be put at the end of them with a single semicolon, too.